src/components/PollResultModal/PollResultChart/PollResultChartBar.tsx

```diff
@@ -85,6 +88,49 @@ export const PollResultChartBar = ({
   return <SimpleBarChart data={results} options={options}></SimpleBarChart>;
 };
 
+export function generateColor(index: number, theme: charts.ChartTheme) {
+  const lightThemeColors = [
+    '#0A60FF',
+    '#5B5201',
+    '#7B24FF',
+    '#A20B54',
+    '#1F5C5A',
+    '#9c1a29f9',
+    '#8A3A0F',
+  ];
+  const darkThemeColors = [
+    '#8AB3FF',
+    '#CBB701',
+    '#C29EFF',
+    '#F684BB',
+    '#40BFBD',
+    '#EB8995',
+    '#ED905E',
+  ];
+  if (theme === 'white') {
+    return lightThemeColors[index];
+  } else {
+    return darkThemeColors[index];
+  }
+}
+
+export function createObjectColors(
```

Comment:
Some ideas for improvements:

* These two functions could be moved into the `helpers.ts` file. You could also test them in `helpers.test.ts`.
* Prefer using the `Theme` of the widget instead of the theme from carbon. We don't want to run into problems if the carbon themes might be renamed.
* `lightThemeColors` and `darkThemeColors` can be defined outside of the function. It could also be replaced by a `const themeColors = theme === 'dark' ? darkThemeColors : lightThemeColors;`. (Prefer checking for `dark` so all unknown theme values fallback to the light colors.)
* The `i >= 7` code (a) can easily break (what if someone adds an eighth number to the arrays and forgets to update the condition) (b) won't work for >14 groups. The [Modulo operator `%`](https://stackoverflow.com/a/44776930/3486395) would be perfect here. Did you ever came across it?

I would also like to restructure it slightly to improve the readability:

```ts
const lightThemeColors = [
  ...
];

const darkThemeColors = [
  ...
];

// you can call this as `createAnswersColorScale(results.map((r) => r.group), theme)`
export function createAnswersColorScale(
  answers: string[],
  theme: Theme
): Record<string, string> {
  const themeColors = theme === 'dark' ? darkThemeColors : lightThemeColors;

  return Object.fromEntries(
    groups.map((answer, i) => [answer, themeColors[i % themeColors.length]])
  );
}
```

Using the original theme requires you to do some changes in the component. I would suggest:

```diff
+ const { theme } = useThemeSelection();
+ const carbonTheme = useCarbonTheme();
- const theme = useCarbonTheme();
  
  // TODO: change all usages of `theme` to `carbonTheme`.
```